plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -432,6 +434,16 @@ instance uni ~ DefaultUni => ToBuiltinMeaning uni DefaultFun where
         makeBuiltinMeaning
             (\() -> [] @(Data,Data))
             (runCostingFunOneArgument . paramMkNilPairData)
+    toBuiltinMeaning MkNil =
+        makeBuiltinMeaning
+            nilPlc
+            (runCostingFunOneArgument . paramMkNilData)
```

Comment:
> Strictly this would need its own cost model parameters instead of re-using the ones for `mkNilData`, and that would require us to add new benchmarks for `mkNil`

Yes, I'm just being sloppy here.

> Could we express the meanings of `mkNilData` and `mkNilPairData` directly in terms of `mkNil`?

Probably not. They are very similar, but not to the point of the first two being an instance of the last one.